src/mscorlib/shared/System/Convert.cs

```diff
@@ -258,79 +248,74 @@ public static Object ChangeType(Object value, TypeCode typeCode, IFormatProvider
                 case TypeCode.Object:
                     return value;
                 case TypeCode.DBNull:
-                    throw new InvalidCastException(Environment.GetResourceString("InvalidCast_DBNull"));
+                    throw new InvalidCastException(SR.InvalidCast_DBNull);
                 case TypeCode.Empty:
-                    throw new InvalidCastException(Environment.GetResourceString("InvalidCast_Empty"));
+                    throw new InvalidCastException(SR.InvalidCast_Empty);
                 default:
-                    throw new ArgumentException(Environment.GetResourceString("Arg_UnknownTypeCode"));
+                    throw new ArgumentException(SR.Arg_UnknownTypeCode);
             }
         }
 
         internal static Object DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
         {
-            Contract.Requires(value != null, "[Convert.DefaultToType]value!=null");
+            Debug.Assert(value != null, "[Convert.DefaultToType]value!=null");
             if (targetType == null)
             {
                 throw new ArgumentNullException(nameof(targetType));
             }
             Contract.EndContractBlock();
 
-            RuntimeType rtTargetType = targetType as RuntimeType;
-
-            if (rtTargetType != null)
```

Comment:
This does not do the same thing as what it did before. The operator == is overloaded for Type, and so this will turn into more expensive call than what used to be here.

It probably does the same thing as before, except that it is quite a bit slower.

@AtsushiKan Any thoughts on this?